app/src/test/java/tttTDD/Java/BoardTest.java

```diff
@@ -64,146 +64,175 @@ public void updatingTheSameSpaceWillResultInAnErrorException() {
-        board.updateBoard(2, "X");
-        board.updateBoard(5, "X");
-        board.updateBoard(8, "X");
+        int[] moves = {2, 5, 8};
+
+        updateCellsOnBoard(moves, "X");
 
         Assert.assertTrue(board.hasWinningSet("X"));
     }
 
     @Test
     public void checkIfTheGameHasAWinningSetOnLeftDiagonalOf3x3Board() {
         board = new Board();
-        board.updateBoard(0, "X");
-        board.updateBoard(4, "X");
-        board.updateBoard(8, "X");
+        int[] moves = {0, 4, 8};
+
+        updateCellsOnBoard(moves, "X");
 
         Assert.assertTrue(board.hasWinningSet("X"));
     }
 
     @Test
     public void checkIfTheGameHasAWinningSetOnRightDiagonalOf3x3Board() {
         board = new Board();
-        board.updateBoard(2, "X");
-        board.updateBoard(4, "X");
-        board.updateBoard(6, "X");
+        int[] moves = {2, 4, 6};
+
+        updateCellsOnBoard(moves, "X");
 
         Assert.assertTrue(board.hasWinningSet("X"));
     }
 
     @Test
     public void checkIfTheGameHasAWinningSetOnTopRowOf5x5Board() {
         board = new Board(5);
-        board.updateBoard(0, "X");
-        board.updateBoard(1, "X");
-        board.updateBoard(2, "X");
-        board.updateBoard(3, "X");
-        board.updateBoard(4, "X");
+        int[] moves = {0, 1, 2, 3, 4};
+
+        updateCellsOnBoard(moves, "X");
 
         Assert.assertTrue(board.hasWinningSet("X"));
     }
 
     @Test
     public void checkIfTheGameHasAWinningSetOnTopRowOf2x2Board() {
         board = new Board(2);
+        int[] moves = {0, 1};
 
-        board.updateBoard(0, "X");
-        board.updateBoard(1, "X");
+        updateCellsOnBoard(moves, "X");
+
+        Assert.assertTrue(board.hasWinningSet("X"));
+    }
+
+    @Test
+    public void checkIfTheGameHasAWinningSetOnTopRowOf100x100Board() {
+        int boardSize = 100;
+        board = new Board(boardSize);
+
+        updateCellsOnLargeBoard(boardSize, "X");
+
+        Assert.assertTrue(board.hasWinningSet("X"));
+    }
+
+    @Test
+    public void checkIfTheGameHasAWinningSetOnTopRowOf1000x1000Board() {
+        int boardSize = 1000;
+        board = new Board(boardSize);
+
+        updateCellsOnLargeBoard(boardSize, "X");
 
         Assert.assertTrue(board.hasWinningSet("X"));
     }
 
     @Test
     public void checkIfTheGameHasAWinningSetOnLeftDiagonalOf6x6Board() {
         board = new Board(6);
-        board.updateBoard(0, "X");
-        board.updateBoard(7, "X");
-        board.updateBoard(14, "X");
-        board.updateBoard(21, "X");
-        board.updateBoard(28, "X");
-        board.updateBoard(35, "X");
+        int[] moves = {0, 7, 14, 21, 28, 35};
+
+        updateCellsOnBoard(moves, "X");
 
         Assert.assertTrue(board.hasWinningSet("X"));
     }
 
     @Test
     public void checkIfTheBoardHasNoMoreMoves() {
         board = new Board();
+        int[] moves = {0, 1, 2, 3, 4, 5, 6, 7, 8};
+        String[] markers = {"X", "O", "X", "O", "O", "X", "O", "X", "X"};
 
```

Comment:
Consideration for another day, but since the moves are linked together by array index, a parameter grouping object could serve you well here. Ie. parameters props that change or are used together are served well to live together.

Ex.
```
MoveValues[] moves = [ { Space: 0, Marker: "X" }, { Space: 1, Marker: "O" } ]
```

Reply:
Yeah, extracting out would add a bunch of code. You could have an array of arrays to start, but I think its a good improvement from what it was

There's probably a data structure that you can use without having to handroll your own java objects. Could be `Tuple<int, String>`, or a Pair<int, String>`, but you might have to import a library to get these

Reply:
Ah, that makes sense. Cool to hear you felt the same way. :)